File: scanner/actions/YoutubeAction.js
var exec = require('child_process').exec
const Action = require('./Action')
const pythonExtensionsPath =
  __dirname + '/../../.virtualenv/bin/python ' + __dirname + '/../../python_extensions'

class YoutubeAction extends Action {
  process() {
    const envVars = this.envVars()
    const scriptPath =
      pythonExtensionsPath +
      '/chromecast.py --app youtube --chromecast_ip ' +
      this.config.chromecast_ip +
      ' ' +
      this.card.uri
    exec(scriptPath, {env: envVars}, function(error, stdout, stderr) {
      console.log(stdout, stderr, error)
    })
  }
}

module.exports = YoutubeAction

File: scanner/actions/youtube_action.py
from .action import ChromecastAction
from pychromecast.controllers.youtube import YouTubeController


class YoutubeAction(ChromecastAction):
    def process(self):
        yt = YouTubeController()
        self.chromecast.register_handler(yt)
        yt_id = self.card["uri"].split(":")[0]
        yt.play_video(yt_id, playlist_id=":".join(self.card["uri"].split(":")[1:]) or None)
